features/48-user-profile-discovery/field-based-acl.md

```diff
@@ -197,6 +235,61 @@ The sample request is shown as followed:
 }
 ```
 
+## Record Partial Update
+
+To support record partial update, warnings are added to the response
+interface of `record:update` action to indicate the rejected fields on a
+**non-atomic** save action.
+
+The following is a sample response of a record partial update:
+
+```js
+{
+  "result": [
+    {
+      // record that saved successfully
+      "_access": [
+        { "level": "read", "public": true }
+      ],
+      "_created_at": "2017-06-19T06:06:16.734626Z",
+      "_created_by": "3c25ddff-b6d4-4d1d-8e67-45c3d82c90f5",
+      "_id": "note/1",
+      "_ownerID": "3c25ddff-b6d4-4d1d-8e67-45c3d82c90f5",
+      "_type": "record",
+      "_updated_at": "2017-06-19T06:33:33.081303Z",
+      "_updated_by": "3c25ddff-b6d4-4d1d-8e67-45c3d82c90f5",
+      "content": "hello world",
+      "tags": ["important", "must-read"]
+    },
+    {
+      // record that saved partially
+      "_access": [
+        { "level": "read", "public": true }
+      ],
+      "_created_at": "2017-06-19T06:06:30.125993Z",
+      "_created_by": "3c25ddff-b6d4-4d1d-8e67-45c3d82c90f5",
+      "_id": "note/2",
+      "_ownerID": "3c25ddff-b6d4-4d1d-8e67-45c3d82c90f5",
+      // list of fields rejected from update
+      "_rejected_fields": ["tags"],
```

Comment:
I think this design lacks extensibility.

From the perspective of the Field ACL, the `_rejected_fields` field satisfied the need to tell the client side which fields were ignored and failed to save. But this field will only work on Field ACL. It is likely that, down the road, we have other kinds of soft-error that we need to tell the user for.

Since the intention is to report soft-error to the client side, I think the new field should contains a list of error object indicating what kinds of error occurred. The error object is already deserializable by all SDKs.

Something akin to this:

```
"_warning": [{"code": 999, "message": "field permission denied", "info": {"fields": "tags"}}]
```

I am trying to make the new error [`DeniedArgument`](https://github.com/SkygearIO/skygear-server/pull/390/files#diff-e365aa9ef3a5e4c5505c294774a21ae0) accepted. I think the same kind of error is useful for indicating soft-error.